lib/common/http-client.ts

```diff
@@ -168,7 +168,7 @@ export class HttpClient implements Server.IHttpClient {
 							this.setResponseResult(promiseActions, cleanupRequestData, { err: new Error(HttpClient.STUCK_RESPONSE_ERROR_MESSAGE) });
 						}
 					}, HttpClient.STUCK_RESPONSE_CHECK_INTERVAL);
-					const successful = helpers.isRequestSuccessful(responseData);
+					const successful = helpers.isRequestSuccessful(responseData) || responseData.statusCode === HttpStatusCodes.NOT_MODIFIED;
```

Comment:
Maybe the check `responseData.statusCode === HttpStatusCodes.NOT_MODIFIED` should be inside `isRequestSuccessful` function

Reply:
I've considered this option, but I'm not sure if it is applicable for everyone who would use `isRequestSuccessful` helper method. By definition [only requests between 200 and 300 are success](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success), so it will seem strange to handle 304 inside that method